crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -160,30 +192,44 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
     }
 
     let in_init = checker.path().ends_with("__init__.py");
-    let fix_init = !checker.settings.ignore_init_module_imports;
+    // TODO: find the `__all__` node
+    let dunder_all = None;
```

Comment:
i have work on this part, but it's been cut from this pr

Reply:
I think best-practice would then be to cut all the other `dunder_all`-related code from this PR. Otherwise, it's not really being tested in the PR that introduces it. And who knows -- maybe we never address this TODO, and that code remains dead in the codebase forever.

Reply:
Here's the entirety of what's being requested here:

```diff
diff --git a/crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs b/crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs
index 3a3302e8c..6dda41f8e 100644
--- a/crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs
+++ b/crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs
@@ -18,7 +18,7 @@ use crate::rules::{isort, isort::ImportSection, isort::ImportType};
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
 enum UnusedImportContext {
     ExceptHandler,
-    Init { first_party: bool, dunder_all: bool },
+    Init { first_party: bool },
 }
 
 /// ## What it does
@@ -108,14 +108,7 @@ impl Violation for UnusedImport {
     fn fix_title(&self) -> Option<String> {
         let UnusedImport { name, multiple, .. } = self;
         let resolution = match self.context {
-            Some(UnusedImportContext::Init {
-                first_party: true,
-                dunder_all: true,
-            }) => "Add unused import to __all__",
-            Some(UnusedImportContext::Init {
-                first_party: true,
-                dunder_all: false,
-            }) => "Use a redundant alias",
+            Some(UnusedImportContext::Init { first_party: true }) => "Use a redundant alias",
             _ => "Remove unused import",
         };
         Some(if *multiple {
@@ -147,7 +140,7 @@ fn is_first_party(qualified_name: &str, level: u32, checker: &Checker) -> bool {
 
 /// For some unused binding in an import statement...
 ///
-///  __init__.py ∧ 1stpty → safe,   move to __all__ or convert to explicit-import
+///  __init__.py ∧ 1stpty → safe,   convert to explicit-import
 ///  __init__.py ∧ stdlib → unsafe, remove
 ///  __init__.py ∧ 3rdpty → unsafe, remove
 ///
@@ -204,8 +197,6 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
 
     let in_init = checker.path().ends_with("__init__.py");
     let fix_init = checker.settings.preview.is_enabled();
-    // TODO: find the `__all__` node
-    let dunder_all = None;
 
     // Generate a diagnostic for every import, but share fixes across all imports within the same
     // statement (excluding those that are ignored).
@@ -234,7 +225,6 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
                             level,
                             checker,
                         ),
-                        dunder_all: dunder_all.is_some(),
                     })
                 } else {
                     None
@@ -265,7 +255,6 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
                     checker,
                     import_statement,
                     to_reexport.iter().map(|(binding, _)| binding),
-                    dunder_all,
                 )
                 .ok(),
             )
@@ -378,13 +367,12 @@ fn fix_by_removing_imports<'a>(
     )
 }
 
-/// Generate a [`Fix`] to make bindings in a statement explicit, either by adding them to `__all__`
-/// or changing them from `import a` to `import a as a`.
+/// Generate a [`Fix`] to make bindings in a statement explicit by adding a redundant alias (e.g.,
+/// converting `import a` to `import a as a`).
 fn fix_by_reexporting<'a>(
     checker: &Checker,
     node_id: NodeId,
     imports: impl Iterator<Item = &'a ImportBinding<'a>>,
-    dunder_all: Option<NodeId>,
 ) -> Result<Fix> {
     let statement = checker.semantic().statement(node_id);
 
@@ -395,10 +383,7 @@ fn fix_by_reexporting<'a>(
         bail!("Expected import bindings");
     }
 
-    let edits = match dunder_all {
-        Some(_dunder_all) => bail!("Not implemented: add to dunder_all"),
-        None => fix::edits::make_redundant_alias(member_names.iter().map(AsRef::as_ref), statement),
-    };
+    let edits = fix::edits::make_redundant_alias(member_names.iter().map(AsRef::as_ref), statement);
 
     // Only emit a fix if there are edits
     let mut tail = edits.into_iter();
```

I leave it to you as to whether you apply the patch.

Reply:
Applied.